### DISCUSSION TITLE
Automazione temperatura e condizione oraria

### DISCUSSION THREAD

**Discussion by gigideveloper** [indomus/forum#1212] [GENERALE - Discussioni generiche]
Salve a tutti ho alcuni problemi con questa automazione che non vuole proprio funzionare.
Il risultato che voglio raggiungere è questo:
Quando all'interno della stanza (in cui vi è un sensore di temperatura Sonoff zigbee) la temperatura scende al di sotto di 20.3 gradi voglio che si attivi una stufetta (collegata ad una preso smart tapo) che riporti la temperatura a 21.5 gradi e dopodichè si spenga una volta raggiunta quella temperatura.
Tutto questo, deve accadere tra le 18:00 e le 07:00 (ovvero nella fascia pomeriggio - notte).
Il problema è che l'automazione non parte automaticamente, ma, se la attivo io manualmente la prima volta ad esempio alle 18:30, funziona correttamente.
Non mi spiego perchè la devo avviare io la prima volta e non si attiva da sola.
Vi posto il codice dell'automazione e il debug.

Automazione

`  alias: Stufetta on
  description: ''
  trigger:
  - platform: numeric_state
    entity_id: sensor.temperature_9
    below: '20.3'
  condition:
  - condition: or
    conditions:
    - condition: time
      after: '18:00'
    - condition: time
      before: 07:00
  action:
  - type: turn_on
    device_id: b67cea6781c9ec8166f7d9cedc0b99e8
    entity_id: switch.presa_mamma
    domain: switch
  mode: single`

 Debug

Executed: 19 dicembre 2021, 16:47:43
Result:
result: false
conditions/0
Executed: 19 dicembre 2021, 16:47:43
Result:
after:
  __type: <class 'datetime.time'>
  isoformat: '18:00:00'
now_time:
  __type: <class 'datetime.time'>
  isoformat: '16:47:43.868764'
before:
  __type: <class 'datetime.time'>
  isoformat: '23:59:59.999999'
result: false
conditions/1
Executed: 19 dicembre 2021, 16:47:43
Result:
after:
  __type: <class 'datetime.time'>
  isoformat: '00:00:00'
now_time:
  __type: <class 'datetime.time'>
  isoformat: '16:47:43.868849'
before:
  __type: <class 'datetime.time'>
  isoformat: '07:00:00'
result: false

Da ignorante, mi sembra che il Now time non si aggiorna automaticamente e pertanto non innesca l'automazione.
Comunque, vi chiedo un supporto per capire cosa non và in tutto.
Grazie

**Comment by sky7176** [comment]
Ciao,
prova a guardare questa guida:
https://indomus.it/progetti/domotizzare-il-riscaldamento-autonomo-tramite-contatto-pulito-e-home-assistant-senza-termostato-fisico/
e anche questa:
https://indomus.it/progetti/creare-un-temporizzatore-per-sistemi-clima-su-home-assistant/

Giuseppe

**Reply by gigideveloper** [reply]
Grazie Giuseppe, ho letto diversi progetti e credo di procedere creando un unità clima, mi sembra la più sensata per i miei scopi.
Anche se non ho risolto il problema perchè non riesco ad innescare l'automazione, l'importante è che raggiungo il risultato.

**Comment by KC-inDomus** [comment]
segui la falsariga [di questo approccio](https://indomus.it/progetti/controllare-un-ventilatore-sulla-base-di-un-range-termico-tramite-home-assistant-v2/), dopodiché una volta definita l'unità climate, usa i suoi servizi con un'automazione legata solo al tempo per attivarla/disattivarla.

**Reply by gigideveloper** [reply]
Ciao ti ringrazio per avermi risposto, effettivamente l'unità climate è tutta un'altra storia e procederò usando questo approccio.
Alla fine credo di aver capito dov'è il problema, ovvero, nel trigger.
Il trigger non innesca l'automazione perchè non riesce in qualche modo a comunicare lo stato attuale della temperatura, ma solo la variazione.
